### PR TITLE
Add HTTPS support

### PR DESCRIPTION
**Pull request by jongio**
Thanks for contribution! Please go through following checklist before sending PR.

### PR Branch Destination

- For Azurite V3, please send PR to `dev` branch.
- For legacy Azurite V2, please send PR to `legacy-dev` branch.

### Always Add Test Cases

> Coming soon.

### Development Guideline

Please go to CONTRIBUTION.md for steps about setting up development environment and recommended Visual Studio Code extensions.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -21,7 +23,28 @@
...
+      "type": "node",
+      "request": "launch",
+      "name": "Azurite Service - Loki, Loose, HTTPS",
+      "cwd": "${workspaceFolder}",
+      "runtimeArgs": ["-r", "ts-node/register"],
+      "args": [
+        "${workspaceFolder}/src/azurite.ts",
+        "-L",
+        "--cert",
+        "C:\\temp\\sslsite\\127.0.0.1.pem",
```
Comment (XiaoningLiu) [inline .vscode/launch.json]: Just thinking if we can have built in cert and keys files for debugging purpose in repo, so these 2 pathes can relatively point to them.

File: README.md [hunk 2]
```diff
@@ -353,9 +353,26 @@ Or customize multi storage accounts and each has 2 keys:
 set AZURITE_ACCOUNTS="account1:key1:key2;account2:key1:key2"
 ```
 
+### HTTP Setup
```
Comment (XiaoningLiu) [inline README.md]: HTTPS?

```suggestion
### HTTPS Setup
```

File: README.md [hunk 3]
```diff
@@ -375,10 +392,47 @@ Or if the SDK or tools support following short connection string:
...
+If you do not set this, then you will get the following error:
+
+```
+unable to verify the first certificate.
+```
+
+Follow these steps to add the HTTPS endpoints to Storage Explorer:
+
+1. Right click on Local & Attached->Storage Accounts and select "Connect to Azure Storage...".
+1. Select "Use a connection string" and click Next.
```
Comment (XiaoningLiu) [inline README.md]:
```suggestion
2. Select "Use a connection string" and click Next.
```

Comment (XiaoningLiu) [reply README.md]: And following indexes

File: src/blob/BlobServerFactory.ts [hunk 4]
```diff
@@ -67,7 +67,9 @@ export class BlobServerFactory {
           undefined,
           debugFilePath !== undefined,
           debugFilePath,
-          env.loose()
+          env.loose(),
```
Comment (XiaoningLiu) [inline src/blob/BlobServerFactory.ts]: Please make same updates for `SqlBlobServer` above.

File: src/common/ServerBase.ts [hunk 5]
```diff
@@ -62,10 +62,14 @@ export default abstract class ServerBase implements ICleaner {
    */
   public getHttpServerAddress(): string {
     const address = this.httpServer.address();
+    const protocol = `http${this.config.hasCert() ? "s" : ""}://`;
+
     if (typeof address === "string" || address === null) {
-      return address || "";
+      return protocol + address;
+    } else if (address === null) {
```
Comment (XiaoningLiu) [inline src/common/ServerBase.ts]: seems not expected. first and second condition both includes `address === null`

File: src/common/VSCEnvironment.ts [hunk 6]
```diff
@@ -74,4 +74,12 @@ export default class VSCEnvironment implements IEnvironment {
   public async debug(): Promise<boolean> {
     return this.workspaceConfiguration.get<boolean>("debug") || false;
   }
+
+  public cert(): string | undefined {
+    return this.workspaceConfiguration.get<string>("cert");
```
Comment (XiaoningLiu) [inline src/common/VSCEnvironment.ts]: `VSCEnvironment` reads Visual Studio Code extension settings. It needs additional configurations to add these 2 settings into Azurite VSC extension in `package.json`. There is a extension settings list defined in `package.json`. Need to update extension supported configurations in Readme.md.

I'm also fine if Azurite Visual Studio exenstion doesn't have https support for now.

File: .vscode/launch.json (hunk 1 above)
Comment (jongio) [reply .vscode/launch.json]: Were you thinking this?
1. Each dev creates their own certs
2. Puts them in the root of the project
3. Update this setting to ${workspaceFolder}/src/127.0.01.pem (same for key)
4. Add those files to the .gitignore so they don't get checked in.

Comment (jongio) [reply .vscode/launch.json]: @XiaoningLiu

Comment (XiaoningLiu) [reply .vscode/launch.json]: @jongio sorry to miss this. Sounds good to me.

**Pull request merged**